Sparkle/SUAppcast.m

```diff
@@ -115,6 +98,31 @@ - (void)downloadDidFinish:(NSURLDownload *)__unused aDownload
     }
 }
 
+- (NSDictionary *)attributesOfNode:(NSXMLElement *)node
+{
+    NSEnumerator *attributeEnum = [[node attributes] objectEnumerator];
+    NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+
+    for (NSXMLNode *attribute in attributeEnum) {
+        NSString *attrName = [self sparkleNamespacedNameOfNode:attribute];
+        if (!attrName) {
+            continue;
+        }
+        dictionary[attrName] = [attribute stringValue];
+    }
+    return dictionary;
+}
+
+-(NSString *)sparkleNamespacedNameOfNode:(NSXMLNode *)node {
+    // XML namespace prefix is semantically meaningless, so compare namespace URI
+    if ([[node URI] isEqualToString:@"http://www.andymatuschak.org/xml-namespaces/sparkle"]) {
+        NSString *localName = [node localName]; assert(localName);
```

Comment:
I've meant to have the assert (I've put it there, since non-nullable annotations caused warnings).

Namespace URI in XML is just a magic string compared codepoint-by-codepoint, never used to fetch anything, so we'll never change it. The fact that everyone settled on using http:// for NS URIs just has to be added to a long list of design problems with namespaces…